aws-lc-rs/src/ec/key_pair.rs

```diff
@@ -224,3 +290,57 @@ fn compute_ecdsa_signature<'a>(
 
     Ok(&mut signature[0..out_sig_len])
 }
+
+/// Elliptic curve private key.
+pub struct PrivateKey<'a>(&'a EcdsaKeyPair);
+
+impl Debug for PrivateKey<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str(&format!("EcdsaPrivateKey({:?})", self.0.algorithm.id))
+    }
+}
+
+/// Elliptic curve private key data encoded as a big-endian fixed-length integer.
+pub struct EcPrivateKeyBuffer {
+    _priv: (),
+}
+
+/// Elliptic curve private key data encoded as DER.
+pub struct EcPrivateKeyDer {
+    _priv: (),
+}
+
+impl PrivateKey<'_> {
+    /// Exposes the private key encoded as a big-endian fixed-length integer.
+    ///
+    /// For most use-cases, `EcdsaKeyPair::to_pkcs8()` should be preferred.
+    ///
+    /// # Errors
+    /// `error::Unspecified` if serialization failed.
+    pub fn to_buffer(&self) -> Result<Buffer<'static, EcPrivateKeyBuffer>, Unspecified> {
+        unsafe {
+            let buffer = ec::marshal_private_key_to_buffer(
+                self.0.algorithm.id,
+                &self.0.evp_pkey.as_const(),
+            )?;
+            Ok(Buffer::<EcPrivateKeyBuffer>::new(buffer))
+        }
+    }
+
+    /// Encode the private key via DER into bytes.
+    ///
+    /// # Errors
+    /// `error::Unspecified`  if serialization failed.
+    pub fn to_der(&self) -> Result<Buffer<'static, EcPrivateKeyDer>, Unspecified> {
```

Comment:
Just wanted to highlight this as another example of ambiguity around meaning of to `to_der`. EcsdaPrivateKey encode to DER but follow PKCS#8 (v1 or v2) and is explicit with the functions. In this case though `to_der` is serializing only to the specific `PrivateKey` format, specifically from [RFC 5915](https://www.rfc-editor.org/rfc/rfc5915). This is the same structure that is wrapped in the same PKCS#8 structure. Now you could argue that KeyPair type makes sense to use PKCS#8 cause that supports both being present (V2). But it also supports serializing just the public key, so why does `to_der` here use that? Basically I think the intent needs to be more clear for the serialization functions.

Reply:
I agree with your concerns. I'm still thinking about what changes need to be made regarding them. 

Here are a few thoughts I have about what's important regarding the serialization/deserialization methods we provide for our structs:
* Documentation should be unambiguous about the encoding(s) provided or required.
* Method names should be clear about their purpose, but don't need to indicate details of the encoding.
* Method names should be consistent/symmetric within a given struct. For example, the encoding serialized by a `to_der` function should be appropriate for a `from_der` to deserialize the same struct (if such a method exists).
* Structs/Types (e.g., `EcPrivateKeyDer`) should be unambiguous and guide users toward proper usage of serialized data.

Reply:
My latest commit adds/improves the documentation so that the encoding provided by serialization methods is more clear. I also moved the structs that differentiate buffer types to a `signature::encoding` module.  We might also rename the (e.g.) `to_der` methods to more directly indicate the type of encoding provided, but I'm not yet sure that's needed.

This is a one-way-door API decision, so it should be carefully considered.